torch/csrc/stable/ops.h

```diff
@@ -51,6 +52,45 @@ inline Tensor narrow(Tensor& self, int64_t dim, int64_t start, int64_t length) {
   return Tensor(ret0);
 }
 
+// We expect this to be a stable version of the new_empty op that takes in
+// only dtype information.
+inline Tensor new_empty(
+    const Tensor& self,
+    std::vector<int64_t> size,
+    std::optional<c10::ScalarType> dtype = std::nullopt) {
+  int32_t device_type;
+  TORCH_ERROR_CODE_CHECK(aoti_torch_get_device_type(self.get(), &device_type));
+
+  int32_t device_index;
+  TORCH_ERROR_CODE_CHECK(
+      aoti_torch_get_device_index(self.get(), &device_index));
+
+  // Handle dtype - use input tensor's dtype if not specified
+  int32_t target_dtype;
+  if (dtype.has_value()) {
+    target_dtype = static_cast<int32_t>(dtype.value());
```

Comment:
hmmm lol is this actually stable...this looks to expose a detail on how dtype is encapsulated. Should we/I write that translation layer from the headeronly ScalarType to their corresponding shim aoti_torch_get_dtype... maybe there's no other way to hide this